### ISSUE TITLE
Miss some metrics in airr.tsv file.

### ISSUE THREAD

**Issue by Chenjunjie1996** [liulab-dfci/TRUST4#147]
Here is the command:
perl trust-airr.pl test1_barcode_report.tsv test1_annotate.fa --format barcoderep > airr_rearrangement.tsv

![120C0A47-A148-4a28-9BB1-4A55EE78836E](https://user-images.githubusercontent.com/74704889/179658509-81b259fc-9e09-4225-ad72-05a5b7a8d7d3.png)

As the figure shown, sequence_alignment, germline_alignment, cdr3nt, cdr3aa, and location of cdr1 cdr2 cdr3 are not in airr file.

**Comment by mourisl** [comment]
sequence_alignment, germline_alignment output is implemented in the github repo. I will draft a new release this week. "cdr3nt, cdr3aa, location of cdr1,2,3" are optional fields in the airr file, so I did not output them. In AIRR specification, "junction" is the cdr3 with flanking motif amino acids, so they are kind of equivalent. Hope this helps. 

If you want to generate the "sequence_alignment, germline_alignment" fields directly from trust-airr script, you need to return annotator with option "--airrAlign" to output the two alignment information, and feed that file to the "--airr-align" option of the trust-airr.pl script.

**Comment by Chenjunjie1996** [comment]
Thank you for your answer.
Is the annotator command? 
![Uploading D7198A2A-DE9C-4f59-BD9D-910EB4A981DC.png…]()

**Comment by Chenjunjie1996** [comment]
![D7198A2A-DE9C-4f59-BD9D-910EB4A981DC](https://user-images.githubusercontent.com/74704889/179670775-eccf4635-679c-4ab5-b901-d486673a2d15.png)

**Comment by mourisl** [comment]
Yes. I did not see --airrAlignment option, so you need to use "git clone" to get the github version.

**Comment by Chenjunjie1996** [comment]
Ok, i will try.
Another question is how to  calculate the umi for each contig? I found the umi count is equal to read count  after converting barcode report to 10x.
![image](https://user-images.githubusercontent.com/74704889/179876865-34af22b1-b082-403d-aa61-44520c66b54c.png)

**Comment by mourisl** [comment]
The UMI information is just used for abundance estimation, that's why you see the read count (abundance) equals to umi when using the "--UMI" option.

**Comment by Chenjunjie1996** [comment]
So, When using --UMI option,  read count is an accurate read count value, and represents the abundance, but the umi is just a abundance metric, not a real umi count value in the 10X report?

**Comment by mourisl** [comment]
Sorry for my bad explanation. When using --UMI option, read count will be computed as UMI count instead of the actual read count.

**Comment by Chenjunjie1996** [comment]
Thanks.
so how to calculate the real read count for each assembled contig?

**Comment by mourisl** [comment]
If you want to the abundance using read count, you can run TRUST4 without the UMI information.

**Comment by Chenjunjie1996** [comment]
many thanks
